src/main/resources/assets/js/components/AppPageComponent.jsx

```diff
@@ -0,0 +1,251 @@
+/** @jsx React.DOM */
+
+var React = require("react/addons");
+var AppBreadcrumbsComponent = require("../components/AppBreadcrumbsComponent");
+var AppVersionListComponent = require("../components/AppVersionListComponent");
+var PageComponent = require("../components/PageComponent");
+var TabPaneComponent = require("../components/TabPaneComponent");
+var TaskDetailComponent = require("../components/TaskDetailComponent");
+var TaskViewComponent = require("../components/TaskViewComponent");
+var TogglableTabsComponent = require("../components/TogglableTabsComponent");
+
+var tabs = [
+  {id: "tasks", text: "Tasks"},
+  {id: "configuration", text: "Configuration"}
+];
+
+var AppPageComponent = React.createClass({
+  displayName: "AppPageComponent",
+
+  propTypes: {
+    activeTask: React.PropTypes.object,
+    appVersionsFetchState: React.PropTypes.number.isRequired,
+    model: React.PropTypes.object.isRequired,
+    destroyApp: React.PropTypes.func.isRequired,
+    fetchTasks: React.PropTypes.func.isRequired,
+    fetchAppVersions: React.PropTypes.func.isRequired,
+    onDestroy: React.PropTypes.func.isRequired,
+    onShowTaskDetails: React.PropTypes.func.isRequired,
+    onShowTaskList: React.PropTypes.func.isRequired,
+    onTasksKilled: React.PropTypes.func.isRequired,
+    restartApp: React.PropTypes.func.isRequired,
+    rollBackApp: React.PropTypes.func.isRequired,
+    scaleApp: React.PropTypes.func.isRequired,
+    suspendApp: React.PropTypes.func.isRequired,
+    tasksFetchState: React.PropTypes.number.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      activeViewIndex: 0,
+      activeTabId: tabs[0].id,
+      selectedTasks: {}
+    };
+  },
+
+  handleDestroyApp: function () {
+    this.props.destroyApp();
+    this.onDestroy();
+  },
+
+  handleRestartApp: function () {
+    this.props.restartApp();
+  },
+
+  onTabClick: function (id) {
+    this.setState({
+      activeTabId: id
+    });
+  },
+
+  toggleAllTasks: function () {
+    var newSelectedTasks = {};
+    var modelTasks = this.props.model.tasks;
+
+    // Note: not an **exact** check for all tasks being selected but a good
+    // enough proxy.
+    var allTasksSelected = Object.keys(this.state.selectedTasks).length ===
+      modelTasks.length;
+
+    if (!allTasksSelected) {
+      modelTasks.forEach(function (task) {
+        newSelectedTasks[task.id] = true;
+      });
+    }
+
+    this.setState({selectedTasks: newSelectedTasks});
+  },
+
+  toggleTask: function (task, value) {
```

Comment:
Though I see that TaskListComponent has the function and, I guess, should be using it